### PR TITLE
Split installation tests from other e2e tests.

### PR DESCRIPTION
**Pull request by nick863**
# Description

In this PR we split the installation time tests from unit and e2e tests.
 - Unit tests and e2e tests are running now in editable install
 - Removed the building wheel step from the tests
 - Added new gate, which builds the wheel and then installs it into virtual environment and logging time.
 - Added test for logging, which, however, does not switched on in CI/CD.
 - Re-enable tests on python 3.9 in CI/CD as we do not depend on wheel in tests anymore i.e. we have separate test for creation of a functional wheel.
 
Example query to application insights:
```
let date_of_run=datetime("2024-07-02");
customEvents
| where timestamp > date_of_run
| where customDimensions has 'installation-test'
| extend parsed = parse_json(customDimensions)
| project 
      name,
      run_date=format_datetime(timestamp, 'yyyy-MM-dd'),
      metric = parsed['activity_name'],
      OS = strcat(parsed["OS"], "_", parsed["OS_release"]),
      python_version = tostring(parsed['python_version']),
      install_time_s=todecimal(parsed["install_time_s"])
| summarize performance = avg(install_time_s) by OS, name
| render columnchart with(
    kind=unstacked
)
```
Result:
image

See work item 3342618 for the inotial task and task 3339407 to enable tests on python 3.9. In this PR we are installing wheel with virtual env, which does not have issue mentioned in the bug. 

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **I confirm that all new dependencies are compatible with the MIT license.**
- [X] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/promptflow-evals-e2e-test-azure.yml [hunk 1]
```diff
@@ -14,21 +14,7 @@ env:
   WORKING_DIRECTORY: ${{ github.workspace }}/src/promptflow-evals
 
 jobs:
-  build:
```
Comment (ninghu) [inline .github/workflows/promptflow-evals-e2e-test-azure.yml:17]: Don't we need the same change for other evals pipelines, like regression test and e2e-test-local?

File: .github/workflows/promptflow-evals-installation-test.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: promptflow-evals-installation-test
```
Comment (ninghu) [inline .github/workflows/promptflow-evals-installation-test.yml:1]: Can we use this chance to rename regression-test to performance-test? Also, I sort of remember that we already capture the installation time in regression test. Can we remove the code there if that's duplicated with the change here?

File: .github/workflows/promptflow-evals-e2e-test-azure.yml (hunk 1 above)
Comment (nick863) [reply .github/workflows/promptflow-evals-e2e-test-azure.yml:17]: Yes, I have removed this code in all other tests, so now they all use editable install.

File: .github/workflows/promptflow-evals-installation-test.yml (hunk 2 above)
Comment (nick863) [reply .github/workflows/promptflow-evals-installation-test.yml:1]: Renamed and removed the installation -time code from performance test.

**Review by singankit — DISMISSED** [review]

**Review by ninghu — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**